Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -628,6 +640,60 @@ void GameList::OpenGCSaveFolder()
     ModalMessageBox::information(this, tr("Information"), tr("No save data found."));
 }
 
+void GameList::AddShortcutToDesktop()
+{
+  const auto game = GetSelectedGame();
+  std::string game_name = game->GetLongName();
+  std::string file_path = game->GetFilePath();
+  std::string file_name = game->GetFileName();
+  std::wstring dolphin_path = QCoreApplication::applicationFilePath().toStdWString();
+  LPCWSTR dolphin_path_convert = dolphin_path.c_str();
+
+  // Arguments
+  std::string args = "-e \"" + file_path + "\"";
+  std::wstring args_convert = std::wstring(args.begin(), args.end());
+
+  // Gets the desktop position with shorcut
+  WCHAR* desktop = 0;
+  SHGetKnownFolderPath(FOLDERID_Desktop, KF_FLAG_NO_ALIAS, NULL, &desktop);
+  CoTaskMemFree(desktop);
+  // Converts desktop path to string
+  std::wstring ws = std::wstring(desktop);
+  std::string desktop_path = std::string(ws.begin(), ws.end());
+  desktop_path = desktop_path + "\\" + game_name + ".lnk";
+  LPCSTR pathLink = desktop_path.c_str();
+
+  HRESULT hres;
+  IShellLink* psl;
+
+  CoInitialize(NULL);
+  hres =
+      CoCreateInstance(CLSID_ShellLink, NULL, CLSCTX_INPROC_SERVER, IID_IShellLink, (LPVOID*)&psl);
+
+  if (SUCCEEDED(hres))
+  {
+    IPersistFile* ppf;
+    LPCWSTR pszArgs = args_convert.c_str();
+
+    psl->SetPath(dolphin_path_convert);
+    psl->SetArguments(pszArgs);
+
+    hres = psl->QueryInterface(IID_IPersistFile, (LPVOID*)&ppf);
```

Comment:
Consider using a different variable here instead of reassigning to the same variable.

Also consider using a C++ style cast here rather than a C-style one

Reply:
If I attempt to use a different variable for QueryInterface, the compiler states that hres_check (a new variable made for QueryInterface) was nullptr.